codec-http/src/main/java/io/netty/handler/codec/http/QueryStringDecoder.java

```diff
@@ -330,7 +331,10 @@ private static String decodeComponent(String s, int from, int toExcluded, Charse
 
             byteBuf.clear();
             do {
-                byteBuf.put(decodeHexByte(s, i, toExcluded));
+                if (i + 3 > toExcluded) {
+                    throw new IllegalArgumentException("unterminated escape sequence at index " + i + " of: " + s);
+                }
+                byteBuf.put(decodeHexDump(s, i + 1, 2));
```

Comment:
I guess either would work in this cause however writing into a `ByteBuf` may be useful if users want to avoid the `byte[]` translation all together ... and this way users don't *have* to know how to properly iterate `hexDump` and the required characteristics.